File: demo/interactive-agent-sdk/typescript/src/configuration.ts
export type DemoConfiguration = Readonly<{
  endpoint: URL;
  token?: string;
  accessContext?: string;
  accessContextHeader: string;
  accessContextEndpoint: URL;
  openAiModel: string;
  verbose: boolean;
}>;

export function optional(name: string): string | undefined {
  const value = process.env[name]?.trim();
  return value ? value : undefined;
}

function required(name: string): string {
  const value = optional(name);
  if (value === undefined) {
    throw new Error(`Missing required TypeScript demo configuration '${name}'.`);
  }
  return value;
}

function flag(name: string): boolean {
  const value = optional(name)?.toLowerCase();
  return value === "1" || value === "true" || value === "yes" || value === "on";
}

export function loadConfiguration(): DemoConfiguration {
  const endpoint = new URL(required("AI_RUNTIME_ENDPOINT"));
  if (!["http:", "https:"].includes(endpoint.protocol)) {
    throw new Error("AI_RUNTIME_ENDPOINT must be an absolute HTTP or HTTPS URL.");
  }

  const configuredAccessContextEndpoint = optional("AI_RUNTIME_ACCESS_CONTEXT_ENDPOINT");
  const accessContextEndpoint = configuredAccessContextEndpoint === undefined
    ? new URL("/auth/access-context", endpoint)
    : new URL(configuredAccessContextEndpoint);

  if (!["http:", "https:"].includes(accessContextEndpoint.protocol)) {
    throw new Error(
      "AI_RUNTIME_ACCESS_CONTEXT_ENDPOINT must be an absolute HTTP or HTTPS URL.",
    );
  }

  const token = optional("AI_RUNTIME_TOKEN");
  const accessContext = optional("AI_RUNTIME_ACCESS_CONTEXT");

  return {
    endpoint,
    ...(token === undefined ? {} : { token }),
    ...(accessContext === undefined ? {} : { accessContext }),
    accessContextHeader:
      optional("AI_RUNTIME_ACCESS_CONTEXT_HEADER") ?? "X-Access-Context",
    accessContextEndpoint,
    openAiModel: required("OPENAI_MODEL"),
    verbose: flag("AI_DEMO_VERBOSE"),
  };
}

export function isSmokeMode(): boolean {
  return process.env.AI_DEMO_SMOKE === "1";
}

File: demo/interactive-agent-sdk/python/src/interactive_agent_sdk_demo/configuration.py
from __future__ import annotations

import os
from dataclasses import dataclass
from urllib.parse import urljoin, urlparse


def optional(name: str) -> str | None:
    value = os.getenv(name)
    if value is None:
        return None
    value = value.strip()
    return value or None


def required(name: str) -> str:
    value = optional(name)
    if value is None:
        raise RuntimeError(f"Missing required Python demo configuration '{name}'.")
    return value


def flag(name: str) -> bool:
    value = (optional(name) or "").lower()
    return value in {"1", "true", "yes", "on"}


def _absolute_http_url(name: str, value: str) -> str:
    parsed = urlparse(value)
    if parsed.scheme not in {"http", "https"} or not parsed.netloc:
        raise RuntimeError(f"{name} must be an absolute HTTP or HTTPS URL.")
    return value


@dataclass(frozen=True)
class DemoConfiguration:
    endpoint: str
    token: str | None
    access_context: str | None
    access_context_header: str
    access_context_endpoint: str
    openai_model: str
    verbose: bool

    @staticmethod
    def load() -> "DemoConfiguration":
        endpoint = _absolute_http_url(
            "AI_RUNTIME_ENDPOINT",
            required("AI_RUNTIME_ENDPOINT"),
        )
        configured_context_endpoint = optional("AI_RUNTIME_ACCESS_CONTEXT_ENDPOINT")
        access_context_endpoint = (
            _absolute_http_url(
                "AI_RUNTIME_ACCESS_CONTEXT_ENDPOINT",
                configured_context_endpoint,
            )
            if configured_context_endpoint is not None
            else urljoin(endpoint, "/auth/access-context")
        )

        return DemoConfiguration(
            endpoint=endpoint,
            token=optional("AI_RUNTIME_TOKEN"),
            access_context=optional("AI_RUNTIME_ACCESS_CONTEXT"),
            access_context_header=(
                optional("AI_RUNTIME_ACCESS_CONTEXT_HEADER")
                or "X-Access-Context"
            ),
            access_context_endpoint=access_context_endpoint,
            openai_model=required("OPENAI_MODEL"),
            verbose=flag("AI_DEMO_VERBOSE"),
        )


def is_smoke_mode() -> bool:
    return os.getenv("AI_DEMO_SMOKE") == "1"
